### DISCUSSION TITLE
Version 1.1.0-RC-1 available for manual download

### DISCUSSION THREAD

**Discussion by darianbjohnson** [Phambili-Tech/Newt_Display#37] [Announcements]
All - you can now download version 1.1.0-RC1 for manual testing. We've added a ton of new features. I'm in the process of writing up the official documentation, but at a high level:
- New API for alerts (beta)
- New API for riddles/brainteasers
- Major change in weather API (new provider and additional weather details)
  - UV
  - Sunrise/sunset
  - Humidity
  - Feels like weather
- Updated settings options:
  - 12/24 Hour clock
  - Nighttime screen mode
  - more Settings details
  - Quote or Weather (or rotating) main screen
  - Imperial or Metric units
- Hello World Arduino example that explains how do:
  - Write to the screen
  - Enable the touchpads
  - Emit a beep from the device

To install manually:

1. Download the latest version of the code (v 1.1.0-Rc-1) here -> https://phambili-pub.s3.amazonaws.com/Newt-1.1.0-RC1.bin
2. Plug your newt into a PC/computer and hit the reset button. A USB Drive should appear (a file named "Firmware.bin" will be in the drive).
3. Drag the file you downloaded (Newt-1.1.0-RC1.binn) onto the USB drive (same location as the Firmware.bin file).
4. Push the reset button

**Comment by rrubanov** [comment]
beautiful. installed.

**Comment by andypiper** [comment]
Works perfectly (so far), here! Picked up the same wifi info, all options appear functional, I'm really impressed with the UI improvements as well. Nice work.

**Comment by badslug** [comment]
Sorry, was away from my Newt but just loaded the beta and everything works and the upgrade was seamless. Loving the new UI, home page, settings and system-wide improvements/tweaks.

One really interesting thing that could be added would be some geeky display of battery charge over time. I'm intentionally trying to run the Newt as long as possible between recharges but if I'm away for the weekend I will typically miss the low battery icon and come back to a dead Newt. It would be interesting to be able to check a post-mortem graph of battery charge over time and "last gasp before death" timestamp so I can better estimate when I need to charge for the next time... also maybe at a low enough charge level the Newt should go into a super low power mode where it never turns on the radios and just shows the current time and low battery warning?

**Reply by darianbjohnson** [reply]
So - I wasn't able to add a "legit" battery fuel monitor to Newt (I couldn't get supply at a decent cost during my manufacturing phase earlier this year). So instead I use a voltage supervisory chip that goes HIGH any time the battery voltage hits 3.6V.

So, I can't chart detailed batter usage... but I do think we can add some stats around:
- average battery life
- time connected to wifi
- time sleep

and also go into "ULTRA" low power mode when I see supervisory pin alerts.... though I built it some buffer (you can go another few days between the initial battery icon going off and the battery dying). So... need to noodle on this a bit. Great ideas

**Reply by badslug** [reply]
Another thought: even if there isn't an actual battery "fuel monitor" capability, maybe you could estimate the battery size based on previous runtimes (or have a setting so you know the battery capacity and can then time the charging times and discharge times) and then just count the amount of lapsed time on battery and then pop up a warning icon a day or two ahead of time with the assumption that the battery runtime will be somewhat consistent. Since you are giving at least a couple days "early warning" these estimates could be pretty crude and still be very useful for normal use-cases. 

Or maybe the opposite, you just have a "days/hours on battery" counter where the battery icon normally sits. I fully admit that with the runtimes I'm getting, I usually forget how long it's been running on battery and unless I randomly decide "hey I've got a charge cord out, let's plug in the Newt" it usually runs out of battery simply due to forgetfulness. At least if I could see it's been running on battery without a charge for over X days (or whatever) then I'd probably charge it more regularly and not have it run out of juice as often as it does.

**Reply by badslug** [reply]
Or maybe there is just a setting of "remind me after X days running on battery to plug Newt in" and when that counter expires then a low battery/plug in battery icon could be displayed.

**Reply by darianbjohnson** [reply]
Great ideas. I logged an issue to track progress - https://github.com/Phambili-Tech/Newt_Display/issues/41
